src/solvers/solver_utils.hpp

```diff
@@ -135,6 +135,165 @@ struct Stencil {
   }
 };
 
+namespace impl { 
```

Comment:
Should there be a `SolverUtils` namespace for these to live in?

Reply:
I changed this to `solvers::utils` instead of `solvers::impl`